pkg/networkservice/common/heal/client_test.go

```diff
@@ -0,0 +1,367 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package heal
```

Comment:
Is it possible to do this testing in package heal_test ?  If so, we probably should, as it prevents us from accidentally doing things that only work because we are in the same package.  Sometimes we *do* need to be in the same package (see the finalizer testing in #83)... but we probably want to only do that if we absolutely must.

Reply:
Almost all tests check the state of private maps (requestors, closers, reported). Moving testing to `heal_test` package make us leave only test that checks calling of `onHeal` after DELETE event.

Reply:
OK... so I guess the question is: what is the functional contract here?  I would maintain it *is* the calling of onHeal after a delete event.  How that happens internally is not actually something we should care about.  I can imagine potentially wanting to test things like:

1.  If we send a DELETE event, do we get the onHeal response promptly?
2. If we send a DELETE event for a non-existent Connection, what happens?
3.  If we send a DELETE event repeatedly, what happens?
4.  If we Close a connection, and then send a DELETE event, what happens?
4.  If we disconnect (and thus send a new INITIAL_STATE) without the Connection, do we get the onHeal response promptly?

etc.

I don't much care about the *internal* state of the chain element... I care about its external interactions :)